frontend/src/component/common/Combo/Combo.tsx

```diff
@@ -0,0 +1,150 @@
+import {
+    HTMLAttributes,
+    forwardRef,
+    useCallback,
+    useRef,
+    useState,
+} from 'react';
+import { Autocomplete, AutocompleteProps, styled } from '@mui/material';
+import { GroupedVirtuoso, GroupedVirtuosoHandle } from 'react-virtuoso';
+
+const VIRTUALIZED_THRESHOLD = 100;
+
+const StyledDiv = styled('div')(({ theme }) => ({
+    'div[data-item-group-index]': {
+        '.MuiAutocomplete-option': {
+            paddingLeft: theme.spacing(3),
+        },
+    },
+}));
+
+interface IComboProps<
+    T,
+    Multiple extends boolean | undefined,
+    DisableClearable extends boolean | undefined,
+    FreeSolo extends boolean | undefined
+> extends AutocompleteProps<T, Multiple, DisableClearable, FreeSolo> {
+    virtualized?: 'auto' | 'always' | 'never' | undefined;
+}
+
+export const Combo = <
+    T,
+    Multiple extends boolean | undefined = false,
+    DisableClearable extends boolean | undefined = false,
+    FreeSolo extends boolean | undefined = false
+>({
+    virtualized = 'auto',
+    ...props
+}: IComboProps<T, Multiple, DisableClearable, FreeSolo>) => {
+    if (
+        virtualized === 'never' ||
+        (virtualized === 'auto' && props.options.length < VIRTUALIZED_THRESHOLD)
+    )
+        return <Autocomplete {...props} />;
+
+    return (
+        <Autocomplete
+            {...props}
+            disableListWrap
+            ListboxComponent={ListboxComponent}
+        />
```

Comment:
Otherwise we use a version with a custom `ListboxComponent` which will allow us to virtualize the list.

We can probably leverage something here to simplify the `ListboxComponent` event handlers (instead of adding to window) and interactions below. I'm thinking maybe adding a ref to the `Autocomplete` and passing it down to the `ListboxComponent`, but needs exploring.